packages/truffle-core/lib/commands/config.js

```diff
@@ -0,0 +1,41 @@
+const command = {
+  command: "config",
+  description: "set user-level configuration options",
+  config: {
+    usage: "truffle config --enable-analytics",
+    options: [   
+      {
+        option: "--enable-analytics",
+        description: "Enable Truffle to send usage data to Google Analytics",
+      }, 
+      {
+        option: "--disable-analytics",
+        description: "Disable Truffle's ability to send usage data to Google Analytics",
+      }
+    ]
+  },
+  builder: {},
+  /**
+  * run config commands to set user-level config settings
+  * @param {Object} options
+  * @param {Func} callback
+  */
+  run: function (options, done) {
+    const googleAnalytics = require("../services/google-analytics");
+    let setAnalytics;
+    if(options.enableAnalytics) {
+      setAnalytics = googleAnalytics.setAnalytics(true);
+    } else if (options.disableAnalytics){
+      setAnalytics = googleAnalytics.setAnalytics(false);
+    } else {
+      setAnalytics = googleAnalytics.setUserConfigViaPrompt();
+      setAnalytics.then(() => done()).catch((err)=> err);
+    }
+
+    if(setAnalytics === true) { 
```

Comment:
You could leave this if statement off and just put a return before the last else's setAnalytics?  On first reading it it feels a little bit cryptic.  Like you could do
```
else {
  setAnalytics = ....
  return setAnalytics.then(() =>... => done());
}
return done();
```

Reply:
I actually had it that way originally, but when the prompt gets run the "return done()" gets called before  it has time to resolve. Setting the if statement avoids that. Is there a different solution maybe?

Reply:
Ahh you are right, I missed that the callback will get called before the async method is done.

Reply:
Yeah, I tried that but then my test throws "TypeError: done is not a function" -- thoughts on how to address? This was a workaround for that issue.